src/scripts/main.js

```diff
@@ -1,3 +1,216 @@
+  </label>
+  <label>
+    Office:
+    <select name="office" data-qa="office" required>
+      <option>Tokyo</option>
+      <option>Singapore</option>
+      <option>London</option>
+      <option>New York</option>
+      <option>Edinburgh</option>
+      <option>San Francisco</option>
+    </select>
+  </label>
+  <label>
+    Age: <input name="age" type="number" data-qa="age" required>
+  </label>
+  <label>
+    Salary: <input name="salary" type="number" data-qa="salary" required>
+  </label>
+  <button type="button">Save to table</button>
+`;
+
+body.append(form);
+
+// Add new row:
+
+form.querySelector('button').addEventListener('click', () => {
+  if (!checkFormFields()) {
+    return;
+  }
+
+  const newRow = document.createElement('tr');
+
+  if (
+    form.elements.name.value &&
+    form.elements.position.value &&
+    form.elements.office.value &&
+    form.elements.age.value &&
+    form.elements.salary.value
+  ) {
+    newRow.insertCell(0).textContent = form.elements.name.value.trim();
+    newRow.insertCell(1).textContent = form.elements.position.value.trim();
+    newRow.insertCell(2).textContent = form.elements.office.value.trim();
+    newRow.insertCell(3).textContent = parseInt(form.elements.age.value, 10);
+
+    const salaryFormatted = parseFloat(
+      form.elements.salary.value.replace(/[^0-9.]/g, ''),
+    );
+
+    newRow.insertCell(4).textContent =
+      `$${new Intl.NumberFormat('us-IN').format(salaryFormatted)}`;
+
+    showNotification(
+      'Success!',
+      'New employee is successfuly added',
+      'success',
+    );
+    form.reset();
+  }
+
+  tBody.append(newRow);
+});
+
+// Appearance of notifications:
+
+function showNotification(title, message, type) {
+  const notification = document.createElement('div');
+
+  notification.className = `notification ${type}`;
+  notification.setAttribute('data-qa', 'notification');
+
+  notification.innerHTML = `
+    <h2 class="title">${title}</h2>
+    <p>${message}</p>
+  `;
+
+  document.body.append(notification);
+
+  setTimeout(() => {
+    notification.remove();
+  }, 4000);
+}
+
+function checkFormFields() {
+  const employeeName = form.elements.name.value.trim();
+
+  if (employeeName.length > 0 && employeeName.length < 4) {
+    showNotification('Error!', 'The minimum name length is 4 letters', 'error');
+
+    return false;
+  }
+
+  const employeeAge = parseInt(form.elements.age.value, 10);
+
+  if (employeeAge < 18 || employeeAge > 90) {
+    showNotification('Error!', 'You are too young/old for this', 'error');
+
+    return false;
+  }
+
+  [...form.elements].every((input) => {
+    if (input.tagName === 'INPUT' || input.tagName === 'SELECT') {
+      if (!input.value.trim()) {
+        showNotification('Error!', 'Fill in all fields', 'error');
+
+        return false;
+      }
+    }
+
+    return true;
+  });
+
+  return true;
+}
+
+cells.forEach((cell) => {
+  const td = document.createElement('td');
```

Comment:
The `td` element is created outside the loop, which means it is reused for all cells. This can cause unexpected behavior. Create a new `td` element inside the event listeners to ensure each cell has its own element.